content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx

```diff
@@ -0,0 +1,530 @@
+
+To remove records which embed a composite type, use the `delete` or `deleteMany` methods. This will also remove the embedded composite type.
+
+For example, use `deleteMany` to delete all products with a `size` of `"Small"`. This will also delete any embedded `photos`.
+
+```ts
+const deleteProduct = await prisma.product.deleteMany({
+  where: {
+    size: 'Small',
+  },
+})
+```
+
+## Filtering composite types with <inlinecode>find</inlinecode> and <inlinecode>findMany</inlinecode>
+
+Composite types can be filtered within a `find` or `findMany` method. The following section describes the methods available for filtering down to a single type or multiple types, and gives examples of each.
+
+### Filter for one composite type
+
+Use the new `is`, `equals` and `isNot` methods to change a single composite type:
+
+- `is`: Filter results by matching composite types. Requires one or more fields to be present _(e.g. Filter orders by the street name on the shipping address)_
+- `equals`: Filter results by matching composite types. Requires all fields to be present. _(e.g. Filter orders by the full shipping address)_
+- `isNot`: Filter results by non-matching composite types.
+
+For example, use `is` to filter for orders with a street name of `'555 Candy Cane Lane'`:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      is: {
+        street: '555 Candy Cane Lane',
+      },
+    },
+  },
+})
+```
+
+You can also use a shorthand notation for this query, where you leave out the `is` and specify just the fields that you want to filter for:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      street: '555 Candy Cane Lane',
+    },
+  },
+})
+```
+
+Use `equals` to filter for orders which match on all fields in the shipping address:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      equals: {
+        street: '555 Candy Cane Lane',
+        city: 'Wonderland',
+        zip: '52337',
+      },
+    },
+  },
+})
+```
+
+Use `isNot` to filter for orders that do not have a `zip` code of `'52337'`:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      isNot: {
+        zip: '52337',
+      },
+    },
+  },
+})
+```
+
+### Filter for many composite types
+
+Use the new `equals`, `isEmpty`, `every`, `some` and `none` methods to filter for multiple composite types:
+
+- `equals`: Checks exact equality of the list.
+- `isEmpty`: Checks if the list is empty.
+- `every`: Every item in the list must match the condition.
+- `some`: One or more of the items in the list must match the condition.
+- `none`: None of the items in the list can match the condition.
+
+For example, you can use `equals` to find products with
+
+```ts
+const product = prisma.product.findMany({
+  where: {
+    equals: {
+      photos: [{ url: '1.jpg' }, { url: '2.jpg' }],
+    },
+  },
+  orderBy: {
```

Comment:
Just a note that we just introduced a new concept here: ordering by composite type.

I'm not sure how best to introduce this in the docs, but we should probably mention it somewhere.

Reply:
Commented out for now.